localization/src/main/kotlin/edu/artic/localization/LanguageSelector.kt

```diff
@@ -44,13 +44,16 @@ class LanguageSelector(private val prefs: LocalizationPreferences) {
      * languages that have been normalized through [Locale.forLanguageTag]. We
      * cannot necessarily trust the original Strings returned by the API.
      */
-    fun <T : SpecifiesLanguage> selectFrom(languages: List<T>): T {
+    fun <T : SpecifiesLanguage> selectFrom(languages: List<T>, prioritizeTour: Boolean = false): T {
         val appLocale: Locale = appLocaleRef.get()
-
         // TODO: Investigate possibility of replacing the list with a `LocaleList`
         return languages.firstOrNull {
             // This is very much intentional. Read the method docs fully before changing.
-            it.underlyingLocale().language == appLocale.language
+            if (prioritizeTour) {
+                it.underlyingLocale().language == prefs.tourLocale.language
+            } else {
+                it.underlyingLocale().language == appLocale.language
+            }
```

Comment:
So, this doesn't quite work. We'll need two loops in some circumstances.

Imagine we are looking at the audio details for one of the stops in a tour. We are working under the environment immediately following:

Available Languages for the audio:
* English
* Chinese

Tour Language selection:
* Spanish

App Language selection:
* Chinese

With the code as given, we will loop through once, not find Spanish, and fall back directly to English. With two loops, we could then search for Chinese and find it successfully.

Reply:
Thank you for detailed explanation @Cliabhach 👍 
Updated it. Please review it.